environment/src/main/java/com/sequenceiq/environment/environment/validation/cloudstorage/EnvironmentBackupLocationValidator.java

```diff
@@ -0,0 +1,36 @@
+package com.sequenceiq.environment.environment.validation.cloudstorage;
+
+import java.util.Optional;
+
+import com.sequenceiq.environment.environment.dto.EnvironmentBackup;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.validation.ValidationResult;
+import com.sequenceiq.environment.environment.domain.Environment;
+
+@Component
+public class EnvironmentBackupLocationValidator {
+
+    private final CloudStorageLocationValidator cloudStorageLocationValidator;
+
+    public EnvironmentBackupLocationValidator(CloudStorageLocationValidator cloudStorageLocationValidator) {
+        this.cloudStorageLocationValidator = cloudStorageLocationValidator;
+    }
+
+    /**
+     * Validate telemetry related logging storage location.
+     * Currently, filter out cloudwatch (or any other cloud logging service) related validations
+     */
+    public ValidationResult validateBackupStorageLocation(Environment environment) {
+        ValidationResult.ValidationResultBuilder resultBuilder = new ValidationResult.ValidationResultBuilder();
+        Optional.ofNullable(environment.getBackup())
+                .filter(backup -> isCloudStorageEnabled(backup) && backup.getCloudwatch() == null)
+                .map(EnvironmentBackup::getStorageLocation)
+                .ifPresent(location -> cloudStorageLocationValidator.validateBackup(location, environment, resultBuilder));
+        return resultBuilder.build();
+    }
+
+    private boolean isCloudStorageEnabled(EnvironmentBackup backup) {
+        return backup.getS3() != null || backup.getAdlsGen2() != null;
```

Comment:
gcs?

Reply:
Because I am only separating from logging bucket to backup bucket, I will not introduce gcs changes as it is not there for logging bucket validation now. 

In the com.sequenceiq.environment.environment.validation.cloudstorage.EnvironmentLogStorageLocationValidator file, we have same thing that was committed by @oleewere. The purpose of this jira is to separate backup from logging bucket but keeps everything as of now.  

    _private boolean isCloudStorageEnabled(EnvironmentLogging logging) {
        return logging.getS3() != null || logging.getAdlsGen2() != null;
    }_

Reply:
yeah, good point, I need to include gcs there